### PR TITLE
#99186 zaken api plugin zaakeigenschappen

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: zgw/catalogi-api/src/main/kotlin/com/ritense/catalogiapi/CatalogiApiPlugin.kt [hunk 1]
```diff
@@ -271,6 +273,27 @@ class CatalogiApiPlugin(
         return results
     }
 
+    fun getEigenschappen(zaakTypeUrl: URI): List<Eigenschap> {
+        var currentPage = 1
+        var currentResults: Page<Eigenschap>?
+        val results = mutableListOf<Eigenschap>()
+
+        do {
```
Comment (ivo-ritense) [inline zgw/catalogi-api/src/main/kotlin/com/ritense/catalogiapi/CatalogiApiPlugin.kt]: Maybe use Page.getAll

File: zgw/catalogi-api/src/main/kotlin/com/ritense/catalogiapi/CatalogiApiPlugin.kt [hunk 2]
```diff
@@ -271,6 +273,27 @@ class CatalogiApiPlugin(
         return results
     }
 
+    fun getEigenschappen(zaakTypeUrl: URI): List<Eigenschap> {
+        var currentPage = 1
```
Comment (theo-ritense) [inline zgw/catalogi-api/src/main/kotlin/com/ritense/catalogiapi/CatalogiApiPlugin.kt]: Please take a look at `Page.getAll(...)`

File: zgw/catalogi-api/src/main/kotlin/com/ritense/catalogiapi/CatalogiApiPlugin.kt (hunk 1 above)
Comment (Klaas-Ritense) [reply zgw/catalogi-api/src/main/kotlin/com/ritense/catalogiapi/CatalogiApiPlugin.kt]: Ye

File: zgw/catalogi-api/src/main/kotlin/com/ritense/catalogiapi/CatalogiApiPlugin.kt (hunk 2 above)
Comment (Klaas-Ritense) [reply zgw/catalogi-api/src/main/kotlin/com/ritense/catalogiapi/CatalogiApiPlugin.kt]: Ok

File: zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/client/ZakenApiClient.kt [hunk 3]
```diff
@@ -448,6 +456,127 @@ class ZakenApiClient(
...
+    fun createZaakeigenschap(
+        authentication: ZakenApiAuthentication,
+        baseUrl: URI,
+        request: CreateZaakeigenschapRequest,
+    ): ZaakeigenschapResponse {
+        validateUrlHost(baseUrl, request.zaak)
+        val result = webclientBuilder
+            .clone()
+            .filter(authentication)
+            .build()
```
Comment (theo-ritense) [inline zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/client/ZakenApiClient.kt]: Can we make a private method for this part? It's being repeated over and over.

Comment (Klaas-Ritense) [reply zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/client/ZakenApiClient.kt]: added

**Review by theo-ritense — APPROVED** [review]

**Pull request merged**